response_operations_ui/static/js/read-csv.js

```diff
@@ -0,0 +1,99 @@
+
+
+
+function checkSampleContents(){
+
+}
+
+
+
+function handleFiles(files) {
+	// Check for the various File API support.
+	if (window.FileReader) {
+		// FileReader are supported.
+		getAsText(files[0]);
+	} else {
+		alert('FileReader is not supported in this browser.');
+	}
+}
+
+function getAsText(fileToRead) {
+	var reader = new FileReader();
+	// Handle errors load
+	reader.onload = loadHandler;
+	reader.onerror = errorHandler;
+	// Read file into memory as UTF-8
+	reader.readAsText(fileToRead);
+}
+
+function loadHandler(event) {
+	var csv = event.target.result;
+	processData(csv);
+}
+
+function processData(csv) {
+    var allTextLines = csv.split(/\r\n|\n/);
+    var lines = [];
+    	while (allTextLines.length) {
+				//lines.push(allTextLines.shift().split(','));
+				lines.push(allTextLines.shift().split(':'));
+    		}
+				//console.log(lines);
+				drawOutput(lines);
+}
+
+
+function errorHandler(evt) {
+	if(evt.target.error.name == "NotReadableError") {
+		alert("Cannot read file :(");
+	}
+}
+
+function drawOutput(lines){
+
+	// Put the form types into their own separate array, so we can interrogate it faster
+	var formTypes = [];
+	for (var i = 0; i < lines.length; i++) {
+			formTypes.push(lines[i][lines.length - 2]);
+	}
+
+	//console.log(formTypes);
+
+	var ciCount = formTypes.filter(function(val, i, arr) {
+	    return arr.indexOf(val) === i;
+	}).length;
+
+
+	//Clear previous data
+	document.getElementById("sample-preview").innerHTML = "";
+
+	var preview = "";
+	preview = preview + "<div class='panel panel--info'>"
```

Comment:
This all works fine so I think it's okay but directly injecting html like this feels a little bit messy

Reply:
I kind of agree, but i'm just leaving it works fine.